File: ocaml/dot_product.ml
let inputs = [1.0; 2.0; 3.0; 2.5];;
let weights = [[0.2; 0.8; -0.5; 1.0]; [0.5; -0.91; 0.26; -0.5]; [-0.26; -0.27; 0.17; 0.87]];;
let biases = [2.0; 3.0; 0.5];;

let rec print_float_list = function 
[] -> ()
| e::l -> print_float e ; print_string " " ; print_float_list l;;

let dot_product v1 v2 = List.fold_left (+.) 0. (List.map2 ( *. ) v1 v2);;
let outputs = List.map2 (fun w b -> dot_product w inputs +. b) weights biases;;

print_float_list outputs;;

File: fortran/dot_product.f90
program DotProduct
	implicit none
	real, dimension(4) :: inputs = [1.0, 2.0, 3.0, 2.5]
	real, dimension(3,4) :: weights = reshape([0.2, 0.8, -0.5, 1.0, 0.5, -0.91, 0.26, -0.5, -0.26, -0.27, 0.17, 0.87], [3,4])
	real, dimension(3) :: biases = [2.0, 3.0, 0.5]
	real, dimension(3) :: outputs
	integer :: i

	outputs = matmul(weights,inputs) + biases

	do i = 1, size(outputs)
		print*, outputs(i)
	end do
end program DotProduct
